ruma-api-macros/src/api.rs

```diff
@@ -163,18 +155,27 @@ impl ToTokens for Api {
             TokenStream::new()
         };
 
-        let extract_request_body =
-            if self.request.has_body_fields() || self.request.newtype_body_field().is_some() {
-                quote! {
-                    let request_body: <RequestBody as ::ruma_api::Outgoing>::Incoming =
-                        ::ruma_api::try_deserialize!(
-                            request,
-                            ::ruma_api::exports::serde_json::from_slice(request.body().as_slice())
-                        );
-                }
+        let extract_request_body = if self.request.has_body_fields()
+            || self.request.newtype_body_field().is_some()
+        {
+            let body_lifetimes = if self.request.has_body_lifetimes() {
+                // duplicate the anonymous lifetime as many times as needed
+                let anon = quote! { '_, };
+                let lifetimes = vec![&anon].repeat(self.request.body_lifetime_count());
+                quote! { < #( #lifetimes )* >}
```

Comment:
It makes more sense to me to have the comma as part of the expansion, not as part of `anon`. Also, you don't need to create a `Vec` here I think. `iter::repeat(quote! { '_ }).take(self.request.body_lifetime_count())` should work just as well (I think).